src/pages/workspace/WorkspacesListRow.tsx

```diff
@@ -0,0 +1,170 @@
+import React, {useMemo} from 'react';
+import {View} from 'react-native';
+import {SvgProps} from 'react-native-svg';
+import {ValueOf} from 'type-fest';
+import Avatar from '@components/Avatar';
+import Icon from '@components/Icon';
+import * as Illustrations from '@components/Icon/Illustrations';
+import Text from '@components/Text';
+import ThreeDotsMenu from '@components/ThreeDotsMenu';
+import withCurrentUserPersonalDetails, {WithCurrentUserPersonalDetailsProps} from '@components/withCurrentUserPersonalDetails';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
+import {AvatarSource} from '@libs/UserUtils';
+import CONST from '@src/CONST';
+import {MenuItem} from '@src/types/onyx/OnyxCommon';
+
+type WorkspacesListRowProps = WithCurrentUserPersonalDetailsProps & {
+    /** Name of the workspace */
+    title: string;
+
+    /** Account ID of the workspace's owner */
+    ownerAccountID: number;
+
+    /** Type of workspace. Type personal is not valid in this context so it's omitted */
+    workspaceType: typeof CONST.POLICY.TYPE.FREE | typeof CONST.POLICY.TYPE.CORPORATE | typeof CONST.POLICY.TYPE.TEAM;
+
+    /** Icon to show next to the workspace name */
+    workspaceIcon?: AvatarSource;
+
+    /** Icon to be used when workspaceIcon is not present */
+    fallbackWorkspaceIcon?: AvatarSource;
+
+    /** Items for the three dots menu */
+    menuItems: MenuItem[];
```

Comment:
You can adjust this type once `MenuItem` [PR](https://github.com/Expensify/App/pull/31141) is merged

Reply:
Updated